lib/Froxlor/Api/Commands/Mysqls.php

```diff
@@ -49,13 +49,23 @@ public function add()
 
 			// parameters
 			$dbserver = $this->getParam('mysql_server', true, 0);
-			$databasedescription = $this->getParam('description', true, '');
+			$databasedescription = $originaldatabasedescription = $this->getParam('description', strtoupper(Settings::Get('customer.mysqlprefix')) == 'DBNAME' ? false : true, '');
 			$sendinfomail = $this->getBoolParam('sendinfomail', true, 0);
 
 			// validation
 			$password = \Froxlor\Validate\Validate::validate($password, 'password', '', '', array(), true);
 			$password = \Froxlor\System\Crypt::validatePassword($password, true);
-			$databasedescription = \Froxlor\Validate\Validate::validate(trim($databasedescription), 'description', '', '', array(), true);
+			if(strtoupper(Settings::Get('customer.mysqlprefix')) == 'DBNAME') {
+			    //If DBNAME is used for prefix, we add customername + id + fieldescription to the validator.
+                //Also we pass in a regex, so we prevent invalid dbnames for mysql.
+                //read more @ https://dev.mysql.com/doc/refman/8.0/en/identifiers.html
+                $databasedescription = \Froxlor\Validate\Validate::validate($GLOBALS['userinfo']['loginname'] . $GLOBALS['userinfo']['customerid'] . '_' . trim($databasedescription), 'description', '/^[0-9a-zA-Z$_]{1,64}$/', '', array(), true);
```

Comment:
$GLOBALS ???? Really??? Please use `$this->getUserDetail('FIELD')`

Also, why additionally put the customer-id in there? Loginame is unique. The database name cannot be any length, there are restrictions

Reply:
The 64 char length is done via the Regexp.
I'll take care of not using GLOBALS - I did not look through the other sourcecode beforehand and chose to just use GLOBALS to get the PR out and then probably get shown the correct way.
I will remove the customer-id.